### PR TITLE
Replace cmake with cc

### PR DESCRIPTION
**Pull request by Jake-Shadle**
This PR replaces the usage of cmake with just straight usage of cc. It replicates some of the logic in the cmakelists but is simplified since this crate only builds a static lib.

Resolves: #51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,18 +16,18 @@ static GLOBAL: MiMalloc = MiMalloc;
...
-By default this library builds mimalloc in secure mode. This add guard pages, 
+By default this library builds mimalloc in secure mode. This add guard pages,
 randomized allocation, encrypted free lists, etc. The performance penalty is usually
 around 10% according to [mimalloc](https://github.com/microsoft/mimalloc)
 own benchmarks.
 
 To disable secure mode, put in `Cargo.toml`:
-```rust
+
+```ini
```
Comment (maroider) [inline README.md:30]:
```suggestion
```toml
```

Comment (Jake-Shadle) [reply README.md:30]: This was actually intentional, some markdown renderers don't actually handle the `toml` code fence, whereas `ini` is usually recognized and tends to be highlighted correctly, at least for simpler code blocks such as this, but I can of course change it to be toml.

Comment (maroider) [reply README.md:30]: Huh, TIL. I have no stake in this either way. It just struck me as odd.

**Comment by repi** [comment]
Possible to get this merged in and publish a new release? Thanks!

**Comment by repi** [comment]
@octavonce any thoughts on getting this in? think it would be beneficial to most and we would be really happy getting rid of our fork

File: libmimalloc-sys/build.rs [hunk 2]
```diff
@@ -1,168 +1,64 @@
...
+    }
 
-    let mut out_dir = "./build".to_string();
-    if cfg!(all(windows)) {
-        if target_env == "msvc" {
-            out_dir.push('/');
-            out_dir.push_str(win_folder);
-        }
+    // Remove heavy debug assertions etc
+    build.define("MI_DEBUG", "0");
```
Comment (thomcc) [inline libmimalloc-sys/build.rs:57]: We should eventually have an option to turn this on. That said, we should set `build.define("NDEBUG", None)` here too if we're doing this. Note that `cmake` will do this automatically, but `cc` does not.

File: libmimalloc-sys/build.rs [hunk 3]
```diff
@@ -1,168 +1,64 @@
...
+
+    let target_os = env::var("CARGO_CFG_TARGET_OS").expect("target_os not defined!");
 
     if cfg!(feature = "override") {
-        cfg = cfg.define("MI_OVERRIDE", "ON");
-    } else {
-        cfg = cfg.define("MI_OVERRIDE", "OFF");
+        // Overriding malloc is only available on windows in shared mode, but we
+        // only ever build a static lib.
+        if target_os != "windows" {
```
Comment (thomcc) [inline libmimalloc-sys/build.rs:35]: This should check the target family, not target_os

**Review by thomcc — CHANGES_REQUESTED** [review]
So, I was working on this at one point and got too busy to continue (and forgot), so I'm thrilled to see it picked up. The version I had was more complex since it kept the `cmake` support under a flag, but I think this is better.

That said, there are a few nits I have.

The most important one is probably mimicing MI_BUILD_OBJECT rather than MI_BUILD_STATIC, since it has many upsides and basically no downsides compared to what we're doing here.

File: libmimalloc-sys/build.rs [hunk 4]
```diff
@@ -1,168 +1,64 @@
...
+        ]
+        .iter()
+        .map(|fname| format!("c_src/mimalloc/src/{}.c", fname)),
+    );
+
+    build.define("MI_STATIC_LIB", None);
+
+    let target_os = env::var("CARGO_CFG_TARGET_OS").expect("target_os not defined!");
 
     if cfg!(feature = "override") {
```
[the remark this answers is not in the document]
Comment (thomcc) [reply libmimalloc-sys/build.rs:32]: (Actually, probably best to address that separately)

File: libmimalloc-sys/build.rs (hunk 2 above)
Comment (thomcc) [reply libmimalloc-sys/build.rs:57]: Hmm, while mimalloc does use NDEBUG, it's not for assertions and just for a couple extra things. I'll file an issue with mimalloc, what we have here is fine actually.

**Comment by thomcc** [comment]
Here's a (tested) patch containing all my suggestions: https://gist.github.com/thomcc/3a499f95c68ad8c8698f61235b1a0fa5

**Pull request merged**